keel-sql/src/main/resources/db/changelog/20200319-resource-artifact-table.yml

```diff
@@ -0,0 +1,25 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-resource-artifact-table
+      author: lpollo
+      changes:
+        - createTable:
+            tableName: resource_artifact
+            columns:
+              - column:
+                  name: resource_uid
+                  type: char(26)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: artifact_uid
+                  type: char(26)
+                  constraints:
+                    nullable: false
+        - addPrimaryKey:
+            tableName: resource_artifact
+            constraintName: res_art_pk
+            columnNames: resource_uid, artifact_uid
```

Comment:
this will still allow multiple resources to use the same artifact, right?
is there anything making sure that each resource can only have one artifact?

Reply:
Yeah, each pair has to be unique, is all. There's nothing in the database preventing a resource from having multiple artifacts. I don't know about the delivery config validation, would have to check. AFAIK, the "only one artifact per resource" assumption was mostly just agreed to verbally, never turned into code.